src/agent.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { Finding, FindingSeverity, FindingType, HandleTransaction, createTransactionEvent } from "forta-agent";
+import agent, { AGENT_REGISTRY_ADDRESS, CREATE_BOT_FUNCTION, DEPLOYER_ADDRESS } from "./agent";
+
+describe("bot created by deployer bot", () => {
+  let handleTransaction: HandleTransaction;
+  const mockTxEventDeployer = createTransactionEvent({
+    transaction: { from: DEPLOYER_ADDRESS.toLocaleLowerCase() },
+  } as any);
+  const mockTxEventStranger = createTransactionEvent({
+    transaction: { from: "0xb0b" },
+  } as any);
+
+  beforeAll(() => {
+    handleTransaction = agent.handleTransaction;
+  });
+
+  describe("handleTransaction", () => {
+    it("returns empty findings if there are no deployments", async () => {
+      mockTxEventDeployer.filterFunction = jest.fn().mockReturnValue([]);
```

Comment:
related to the comment above. we don't mock `filterFunction`
but we add traces to a transaction and test it more realistically.

you can find examples in our repo, inside the projects that use `.filterFunction`

it's like the same for the rest as well.